localnode/localnode.go

```
@@ -0,0 +1,359 @@
+	"math/big"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/cryptoriums/packages/compiler"
+	"github.com/cryptoriums/packages/logging"
+	tx_p "github.com/cryptoriums/packages/tx"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/pkg/errors"
+)
+
+type NodeType string
+
+var (
+	Hardhat NodeType = "hardhat"
+	Anvil   NodeType = "anvil"
+)
+
+const DefaultUrl = "ws://127.0.0.1:8545"
+
+func initAccounts() []tx_p.Account {
+	var Accounts []tx_p.Account
+	for _, addr := range []string{
+		"0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80",
+		"0x59c6995e998f97a5a0044966f0945389dc9e86dae88c7a8412f4603b6b78690d",
+		"0x5de4111afa1a4b94908f83103eb1f1706367c2e68ca870fc3fb9a804cdab365a",
+		"0x7c852118294e51e653712a81e05800f419141751be58f605c371e15141b007a6",
+		"0x47e179ec197488593b187f80a00eb0da91f1b9d0b13f8733639f19c30a34926a",
+		"0x92db14e403b83dfe3df233f83dfa3a0d7096f21ca9b0d6d6b8d88b2b4ec1564e",
+		"0x4bbbf85ce3377467afe5d46f804f221813b2bb87f24d81f60f1fcdbf7cbf4356",
+		"0xdbda1821b80551c9d65939329250298aa3472ba22feea921c0cf5d620ea67b97",
+		"0x2a871d0798f97d79848a013d4936a73bf4cc922c825d33c1cf7073dff6d409c6",
+		"0xf214f2b2cd398c806f84e317254e0f0b801d0643303237d97a22a48e01628897"} {
+
+		acc, err := tx_p.AccountFromPrvKey(addr)
+		if err != nil {
+			panic(err)
+		}
+		Accounts = append(Accounts, acc)
+	}
+
+	return Accounts
+}
+
+type localNode struct {
+	nodeType    NodeType
+	forkNodeURL string
+	nodeURL     string
+	cmd         *exec.Cmd
+	accounts    []tx_p.Account
+	logger      log.Logger
+}
+
+func New(logger log.Logger, nodeType NodeType, forkNodeURL string, blockNumber string) *localNode {
+	if forkNodeURL == "" {
+		logger.Log("invalid forkNodeURL")
+		os.Exit(1)
```

Where possible lets return an error and handle it in the caller

---

Agreed. I was following old pattern here... I'll change